Emulator/BF.File.Emulator/BF.File.Emulator.csproj

```diff
@@ -0,0 +1,58 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0-windows</TargetFramework>
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+    <LangVersion>10.0</LangVersion>
+    <Nullable>enable</Nullable>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+    <OutputPath>$(RELOADEDIIMODS)/reloaded.universal.fileemulationframework.bf</OutputPath>
+    <Deterministic>true</Deterministic>
+    <ImplicitUsings>enable</ImplicitUsings>
+
+    <!-- IL Trimming Support for Framework Dependent Apps. Unofficial.
+         If you set this to true, trimming will be enabled when you publish the mod.
+         Test trimming with: `BuildLinked.ps1` in PowerShell. -->
+    <ReloadedILLink>false</ReloadedILLink>
```

Comment:
Have you tried linking? Given size of libraries, might be beneficial here.

Note: If you have .NET 8 SDK installed, you might need to add `global.json` to force .NET 7 SDK; the current linking script was borked due to internal changes in .NET 8 build targets.

Reply:
I have now (in cd5ddca). 
Trimming Script Compiler caused problems but I was able to trim some dependencies without any issue so that's some size savings.

Reply:
Is the issue JSON serialization? (System.Text.Json)

You'll need to use source generators in that case. It's not documented on MSDN (for some reason), and a Google search will lead you to blog post where the APIs were introduced, but this blog should be a good, very short summary https://blogs.u2u.be/peter/post/using-system-text-json-source-generator-to-improve-performance